mffpy/xml_files.py

```diff
@@ -1025,3 +1025,155 @@ def get_serializable_content(self):
+        return [self._parse_entry(entry) for entry in self.findall('entry')]
+
+    def _parse_entry(self, entry_el):
+        assert self.nsstrip(entry_el.tag) == 'entry', f"""
+        Unknown tool with tag {self.nsstrip(entry_el.tag)}"""
+        tool_el = self.find('tool', entry_el)
+        return {
+            tag: self._entry_type_converter[tag](el)
+            for tag, el in map(lambda e: (self.nsstrip(e.tag), e), tool_el)
+        }
+
+    @classmethod
+    def content(cls, entries: List[dict]) -> dict:  # type: ignore
+        """return content in xml-convertible json format
+
+        Note
+        ----
+        `entries` is a list of dicts with several keys, none of which are
+        required. `entries` should have the following structure:
+        ```
+        entries = [
+            {
+                'name': <str>,
+                'kind': <str>,
+                'method': <str>,
+                'version': <str>,
+                'beginTime': <datetime object>,
+                'endTime': <datetime object>,
+                'sourceFiles': <List[str]>,
+                'settings': <List[str]>,
+                'results': <List[str]>
+            }
+        ]
+        ```
+        """
+        formatted_entries = []
+        for entry in entries:
+            formatted = {}
+            for tag, text in entry.items():
+                assert tag in cls._entry_type_reverter, "entry property "
+                f"'{text}' not serializable. Needs to be one of "
+                f"{list(cls._entry_type_reverter.keys())}."
+                formatted[tag] = {
+                    TEXT: cls._entry_type_reverter[tag](text)  # type: ignore
+                }
+            formatted_entries.append({TEXT: formatted})
+        return {
+            'entry': [
+                {
+                    TEXT: {
+                        'tool': e
+                    }
+                }
+                for e in formatted_entries
+            ]
+        }
+
+    def get_content(self):
+        """return history entries"""
+        formatted_entries = []
+        for entry in self.entries:
+            entry['beginTime'] = self._dump_datetime(entry['beginTime'])
+            entry['endTime'] = self._dump_datetime(entry['endTime'])
+            formatted_entries.append(entry)
+        return formatted_entries
+
+    def get_serializable_content(self):
+        """return a serializable object containing history entries"""
+        return copy.deepcopy(self.get_content())
+
+    def mff_flavor(self) -> str:
+        """return either 'continuous', 'segmented',
+        or 'averaged' representing mff flavor"""
+        segmented = False
+        for entry in self.entries:
+            if entry['method'] == 'Segmentation':
+                segmented = True
+            if entry['method'] == 'Averaging':
+                return 'averaged'
+        if segmented:
+            return 'segmented'
+        else:
+            return 'continuous'
```

Comment:
I know your solution is faster, but this would be easier to understand I think.  I also made it case insensitive in order to avoid these types of problems.
```python
methods = { entry['method'].lower() for entry in self.entries }
if 'averaging' in methods:
    return 'averaged'
elif 'segmentation' in methods:
    return 'segmented'
else:
    return 'continuous'
```

Reply:
Nice! Yeah I was trying to figure out a way to write that to make it a little more intuitive.